main/src/main/scala/sbt/Defaults.scala

```diff
@@ -1769,14 +1776,18 @@ object Classpaths {
                                                  else "release",
                                                logging = ivyLoggingLevel.value),
     deliverConfiguration := deliverLocalConfiguration.value,
-    publishConfiguration := publishConfig(
-      packagedArtifacts.in(publish).value,
-      if (publishMavenStyle.value) None else Some(deliver.value),
-      resolverName = getPublishTo(publishTo.value).name,
-      checksums = checksums.in(publish).value,
-      logging = ivyLoggingLevel.value,
-      overwrite = isSnapshot.value
-    ),
+    publishConfiguration := {
+      // TODO(jvican): I think this is a bug.
```

Comment:
I think this is a bug @eed3si9n. It's not clear to me that we should deliver unconditionally here. The previous if expression has the same semantics as this one, but I think it was covering an underlying misbehaviour.